src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleDictionary.cs

```diff
@@ -46,6 +46,11 @@ private static void HandleStartDictionary(JsonSerializerOptions options, ref Utf
                 }
                 else
                 {
+                    if (classInfo.CreateObject == null)
+                    {
+                        ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(classInfo.Type, reader, state.JsonPath);
+                        return;
+                    }
```

Comment:
Is this the right place to do this test?  Presumably we could tell when a JsonClassInfo was created whether or not it needed a CreateObject delegate, and if it did, do the check then once and throw if necessary, rather than doing the check every time it's needed?

Reply:
This check is added to throw json exception for ```{\"test\": {}}```.

> Presumably we could tell when a JsonClassInfo was created whether or not it needed a CreateObject delegate, and if it did, do the check then once and throw if necessary

yes if its required we set it non-null. JsonClassInfo object is created on the basis of the return type and is unaware of the input string. So we will have to throw while parsing the input.

@steveharter can you suggest a better place of adding this check so that we dont do this check everytime ?

Reply:
We could place several more checks in the "warm-up" code. This disadvantage is that we would always error on a property even though there may never be any JSON for it. However, the user could place a [JsonIgnore] on them to avoid that edge case. If that is acceptable, then I could point you to the location.

Reply:
> This disadvantage is that we would always error on a property even though there may never be any JSON for it.

I don't understand what you mean here. Are you referring to a case like this:
```C#
public class Foo
{
   public Dictionary<string, int[]> Collection {get; set;}
   public string Bar {get; set;}
}
 // This currently doesn't throw since we never try to deserialize Collection
JsonSerializer.Parse("{\"bar\":\"hello\"", typeof(Foo));
```

Maybe we should throw up front, regardless. Otherwise, the user might find it surprising if/when the payload changes that their object model no longer works.

>  However, the user could place a [JsonIgnore] on them to avoid that edge case. If that is acceptable, then I could point you to the location.

I think saving additional runtime checks is a good idea. However, what if the user wants to be able to serialize `Foo`, but ignore the `Collection` property on deserialization only?

Reply:
@ahsonkhan yes -- not all properties may be bound to JSON, some never. However since throwing up-front is best for performance and we have stated previously that we favor performance over edge cases like this.

However I realize that in this case there is just a raw value - so there is no "warm up" here.

So for a case like `JsonSerializer.Parse<int[]>(@"1")` the "1" value is first converted to an integer with value 1 (as expected) and then an attempt to add to a temporary list which doesn't exist (since there were no array brackets in the JSON). So I think this PR is fine for the structure of the current code.